src/test/scala-2/chiselTests/SIntOps.scala

```diff
@@ -284,4 +284,18 @@ class SIntOpsSpec extends AnyPropSpec with Matchers with ShiftRightWidthBehavior
     blit.x.litOption should be(Some(2))
     blit.y.litOption should be(Some(9))
   }
+
+  property("SInt literals with too small of a width should be rejected") {
+    // Sanity checks.
+    0.S.getWidth should be(1)
+    0.S(0.W).getWidth should be(0)
+    -1.S.getWidth should be(1)
+    1.S.getWidth should be(2)
+    // The real check.
+    -2.S.getWidth should be(2)
+    an[IllegalArgumentException] shouldBe thrownBy(-2.S(1.W))
+    0xde.S.getWidth should be(9)
+    an[IllegalArgumentException] shouldBe thrownBy(0xde.S(8.W))
+    an[IllegalArgumentException] shouldBe thrownBy(0xde.S(4.W))
+  }
```

Comment:
These tests look right! I haven't thought through the other ramifications of this change, other than that the existing tests in the repo all pass.

Reply:
Yeah I'm confident it's correct. Note that even before this change `0xde.S.getWidth == 9`, it's just that we were accepting 1-bit less than the correct answer if the user specified it which would then coincidentally do the right thing in _most_ circumstances due to the FIRRTL we emitted actually having the correct width semantics. Thus things would work as long as you didn't introspect on the width and do things with it.